apps/api/src/app/events/usecases/digest-filter-steps/digest-filter-steps-regular.usecase.ts

```diff
@@ -10,16 +10,13 @@ export class DigestFilterStepsRegular {
 
   public async execute(command: DigestFilterStepsCommand): Promise<NotificationStepEntity[]> {
     const steps = [DigestFilterSteps.createTriggerStep(command)];
-    let delayedDigests: JobEntity = null;
+    let delayedDigests;
+
     for (const step of command.steps) {
-      if (step.template.type === StepTypeEnum.DIGEST) {
+      if (step?.template?.type === StepTypeEnum.DIGEST) {
         delayedDigests = await this.getDigest(command, step);
       }
 
-      if (delayedDigests) {
-        continue;
-      }
```

Comment:
This bit with the old `shouldAddDigestUseCase` were creating a double race condition that were mutually dependent.
When creating steps we were checking if there was a delayed digest and when deciding if adding the digest or not we were checking if existed or not.
We have yet to clarify the digest code flow a bit (in a forthcoming PR) but we decided to go through the direction of always creating all the steps for a digest job, no matter what, and at the time of deciding if adding the Digest job or not to the queue, either mark as merged/digested the steps for the digest jobs that are merged.

Reply:
@davidsoderberg cc